tools/status_check.py

```diff
@@ -0,0 +1,138 @@
+# Validate that the framecounts in STATUS.md are up-to-date. This is done by trying to run the entire test case and
+# asserting that the sync framecount is the highest it currently can be. This helps us capture when changes focused
+# on one test case actually benefit a different test case.
+
+from generate_tests import generate_tests
+import json
+import os
+import subprocess
+import sys
+from typing import Dict
+
+STATUS_TEST_CASE_FILENAME = 'statusTestCases.json'
+
+
+class TestCase:
+    def __init__(self, line: str):
+        # Get the test case name in between the square brackets
+        self.name = line[line.find('[') + 2:line.find(']') - 1]
+
+        # Get the frame counts in between the pipes
+        frame_counts = line.split('|')[1].split('/')
+        self.sync_frame = int(frame_counts[0])
+        self.total_frames = int(frame_counts[1])
+
+    # While normally "targetFrame" would just be the lower bound, we want to run the upper bound
+    # and see if it produces a later desync than we currently are tracking in STATUS.md
+    def to_json(self) -> Dict[str, str]:
+        return {
+            "rkgPath": "samples/" + self.name + ".rkg",
+            "krkgPath": "samples/" + self.name + ".krkg",
+            "targetFrame": self.total_frames
+        }
+
+
+class KinokoOutputValidator:
+    def __init__(self, stdout_str):
+        self.test_cases = self.parse_test_cases(stdout_str)
+
+    # e.g. Get "rr-ng-rta-2-24-281" from [TestDirector.cc:112] Test Case
+    # Passed: rr-ng-rta-2-24-281 [1207 / 9060]
+    def parse_test_case_name(self, line: str) -> str:
+        return line.split(':')[2].split(' ')[1]
+
+    # e.g. Get the "814" from [TestDirector.hh:55] Test Case Failed: rr-ng-rta-2-24-281 [814 / 9060]
+    def get_test_case_frame_lower_bound_from_stdout(self, line: str) -> int:
+        return int(line.split(':')[2].split('[')[1].split(' ')[0])
```

Comment:
If `self` isn't used here, these methods could be static. Is it worth not including them as arguments at all, but rather as a member of the class? `self.current_line`?

Reply:
Alright I'm on-board with this. I'll make it a member!